solvebio/query.py

```diff
@@ -852,11 +852,11 @@ def join(self, query_b, key, key_b=None, prefix="b_", always_prefix=True):
 
         # Prepare the filters for the B query expression
         query_params = query_b._build_query()
-        base_filter = '["{}", get(record, "{}")]'.format(key_b, key)
+        # base_filter = '["{}", get(record, "{}")]'.format(key_b, key)
```

Comment:
I think this is the critical part of the join, as far as I remember. Why is it being removed?

Reply:
Moving it into the expression down below makes the join very inefficient as it will pull all records and filter in the annotator as opposed to within Elasticsearch.

Reply:
if base_filter stays, `join()` does not behave as a true left join. For each record, it will query a new dataset according to filters, which is not what we expect to have when the `query_b` has been created.